### PR TITLE
config: Remove deprecated settings for `8.0`

### LINKED ISSUE #5996: Remove deprecated config settings
* `apm-server.jaeger.*`
* `apm-server.instrumentation.* `
* `apm-server.secret_token`
* `apm-server.api_key`

### PR DESCRIPTION
**Pull request by marclop**
## Motivation/summary

Removes all the deprecated config settings, updating the unit and system
tests to use the new non deprecated fields.

## Checklist

- [x] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/master/CHANGELOG.asciidoc)
- [ ] Documentation has been updated

## How to test these changes

1. `docker-compose up -d`
2. `make`
3. Run apm server and verify that `ignoring deprecated config` isn't found in the logs:
```console
$ ./apm-server -e -E output.elasticsearch.username=admin -E output.elasticsearch.password=changeme -E apm-server.instrumentation.enabled=true \
  -E apm-server.rum.allowed_service=\[aservice\] \
  -E apm-server.rum.event_rate.limit=1 \
  -E apm-server.api_key.enabled=true \
  -E apm-server.secret_token=some_secret
```
4. `curl localhost:8200/foo`
5. Via Kibana dev tools verify there are no instrumentation metrics / traces ingested:
```javascript
GET apm-*,traces-apm*,logs-apm*/_search
{
  "query": {
    "match_all": {}
  }
}
```

## Related issues

Closes #5996

### CODE REVIEW

File: systemtest/rum_test.go [hunk 1]
```diff
@@ -87,8 +87,11 @@ func TestRUMAllowServiceNames(t *testing.T) {
 	srv := apmservertest.NewUnstartedServer(t)
 	srv.Config.AgentAuth.SecretToken = "abc123"
 	srv.Config.RUM = &apmservertest.RUMConfig{
-		Enabled:           true,
-		AllowServiceNames: []string{"allowed"},
+		Enabled: true,
+	}
+	srv.Config.AgentAuth.Anonymous = &apmservertest.AnonymousAuthConfig{
+		Enabled:      true,
+		AllowService: []string{"allowed"},
```
Comment (marclop) [inline systemtest/rum_test.go:94]: I've noticed that I needed to set `apm-server.auth.anonymous.enabled=true` in order for `AllowService` to work, while I understand this is intended, should we note this somewhere in our docs? It seems to deviate from what was previously required, just enabling `rum.enabled: true` and `rum.allow_service: [a, b]?

File: beater/config/config.go [hunk 2]
```diff
@@ -62,12 +62,10 @@ type Config struct {
 	Expvar                    ExpvarConfig            `config:"expvar"`
 	Pprof                     PprofConfig             `config:"pprof"`
 	AugmentEnabled            bool                    `config:"capture_personal_data"`
-	SelfInstrumentation       InstrumentationConfig   `config:"instrumentation"`
 	RumConfig                 RumConfig               `config:"rum"`
 	Register                  RegisterConfig          `config:"register"`
 	Kibana                    KibanaConfig            `config:"kibana"`
 	KibanaAgentConfig         KibanaAgentConfig       `config:"agent.config"`
-	JaegerConfig              JaegerConfig            `config:"jaeger"`
```
Comment (axw) [inline beater/config/config.go:70]: oops :sweat_smile: thanks

File: systemtest/rum_test.go (hunk 1 above)
Comment (axw) [reply systemtest/rum_test.go:94]: Sounds good, can you please open a docs issue for that?

**Review by axw — APPROVED** [review]
:tada:

**Comment by stuartnelson3** [comment]
confirmed with https://github.com/elastic/apm-server/commit/2d1c27052b5ddce37858c85b48a84561752f791d

**Pull request merged**